codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,419 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+import io.netty.util.concurrent.FastThreadLocal;
+
+import java.util.BitSet;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+
+/**
+ * A formatter for HTTP header dates, such as "Expires" and "Date" headers, or "expires" field in "Set-Cookie".
+ *
+ * On the parsing side, it honors RFC6265 (so it supports RFC1123).
+ * Note that:
+ * <ul>
+ *     <li>Day of week is ignored and not validated</li>
+ *     <li>Timezone is ignored, as RFC6265 assumes UTC</li>
+ * </ul>
+ * If you're looking for a date format that validates day of week, or supports other timezones, consider using
+ * java.util.DateTimeFormatter.RFC_1123_DATE_TIME.
+ *
+ * On the formatting side, it uses RFC1123 format.
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc6265#section-5.1.1">RFC6265</a> for the parsing side
+ * @see <a href="https://tools.ietf.org/html/rfc1123#page-55">RFC1123</a> for the encoding side.
+ */
+public final class HttpHeaderDateFormatter {
+
+    private static final BitSet DELIMITERS = new BitSet();
+    static {
+        DELIMITERS.set(0x09);
+        for (char c = 0x20; c <= 0x2F; c++) {
+            DELIMITERS.set(c);
+        }
+        for (char c = 0x3B; c <= 0x40; c++) {
+            DELIMITERS.set(c);
+        }
+        for (char c = 0x5B; c <= 0x60; c++) {
+            DELIMITERS.set(c);
+        }
+        for (char c = 0x7B; c <= 0x7E; c++) {
+            DELIMITERS.set(c);
+        }
+    }
+
+    private static final String[] DAY_OF_WEEK_TO_SHORT_NAME =
+            new String[]{"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"};
+
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1161)